src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/HotReload/ProjectHotReloadSession.cs

```diff
@@ -248,10 +254,15 @@ public async ValueTask RestartAsync(CancellationToken cancellationToken)
                 ),
                 cancellationToken);
 
-            await _callback.RestartProjectAsync(cancellationToken);
 
-            // TODO: Should we stop the session here? Or does someone else do it?
-            // TODO: Should we handle rebuilding here? Or do we expect the callback to handle it?
+            if (_callback is HotReloadState hrs)
```

Comment:
What do you think about adding an `internal interface IProjectHotReloadSessionCallback2` (i.e. version 2) and adding the new members to that? Then `HotReloadState` could remain `private`. Instead, here you'd be looking for the newer interface version and using its capabilities when available.

It would also allow us to make the interface public for others to implement (which is presumably the reason `IProjectHotReloadSessionCallback` is public).

Reply:
Sure, one comment though is we probably want to always keep `IProjectHotReloadSessionCallback2` internal. Not sure why `IProjectHotReloadSessionCallback` was made public at first, there are so many project-system specific details in order to implement that interface.

Reply:
It's fine to keep it internal.